engine/robot_logic/traversal.py

```diff
@@ -208,64 +212,98 @@ def update_state(robot_state, velocity, omega):
     return measurements
 
 
-def traverse_roomba(robot_state, base_station_loc, time_limit, roomba_radius):
+def traverse_roomba(robot_state, base_station_loc, time_limit, roomba_radius, database):
     """Move the robot in a roomba-like manner.
     Args:
         base_station_loc (Tuple): The (x,y) location of the base station
         time_limit (Double): How long the robot will move using roomba mode
         roomba_radius (Double): Maximum radius from the base station that the robot can move
     Returns:
-        None
+        robot_state
     """
     # to add obstacle avoidance, can either change roomba to go to point on the circle in the direction of the
     # robot direction or use raw sensor value
     # can we guarantee that when giving a velocity that the robot will move at that velocity?
-    dt = 0
+    dt = 10
+    accumulated_time = 0
     # TODO: battery_limit, time_limit, tank_capacity is full
     exit_boolean = False
-    from electrical.ultrasonic_sensor import Ultrasonic
+    if not robot_state.is_sim:
+        from electrical.ultrasonic_sensor import Ultrasonic
 
-    front_ultrasonic = Ultrasonic(0)
+        front_ultrasonic = Ultrasonic(0)
```

Comment:
don't we already do this in robot setup_logic? Why can't we just directly use the front_ultrasonic in robot_state?